h/static/scripts/annotator/plugin/pdf.coffee

```diff
@@ -1,302 +1,113 @@
-Promise = require('es6-promise').Promise
+extend = require('extend')
+Promise = global.Promise ? require('es6-promise').Promise
 Annotator = require('annotator')
-$ = Annotator.$
 
-detectedPDFjsVersion = PDFJS?.version.split(".").map parseFloat
 
-# Compare two versions, given as arrays of numbers
-compareVersions = (v1, v2) ->
-  unless Array.isArray(v1) and Array.isArray(v2)
-    throw new Error "Expecting arrays, in the form of [1, 0, 123]"
-  unless v1.length is v2.length
-    throw new Error "Can't compare versions in different formats."
-  for i in [0 ... v1.length]
-    if v1[i] < v2[i]
-      return -1
-    else if v1[i] > v2[i]
-      return 1
-  # Finished comparing, it's the same all along
-  return 0
+class PDF extends Annotator.Plugin
+  documentLoaded: null
+  observer: null
+  pdfViewer: null
 
-# Document mapper module for PDF.js documents
-class window.PDFTextMapper extends PageTextMapperCore
-
-  # Are we working with a PDF document?
-  @isPDFDocument: ->
-    PDFView? or             # for PDF.js up to v1.0.712
-      PDFViewerApplication? # for PDF.js v1.0.907 and up
-
-  # Can we use this document access strategy?
-  @applicable: -> @isPDFDocument()
-
-  requiresSmartStringPadding: true
-
-  # Get the number of pages
-  getPageCount: -> @_viewer.pages.length
-
-  # Where are we in the document?
-  getPageIndex: -> @_app.page - 1
-
-  # Jump to a given page
-  setPageIndex: (index) -> @_app.page = index + 1
-
-  # Determine whether a given page has been rendered
-  _isPageRendered: (index) ->
-    @_viewer.pages[index]?.textLayer?.renderingDone
-
-  # Get the root DOM node of a given page
-  getRootNodeForPage: (index) ->
-    @_viewer.pages[index].textLayer.textLayerDiv
-
-  constructor: ->
-    # Set references to objects that moved around in different versions
-    # of PDF.js, and define a few methods accordingly
-    if PDFViewerApplication?
-      @_app = PDFViewerApplication
-      @_viewer = @_app.pdfViewer
+  pluginInit: ->
+    cache = {
```

Comment:
I think it would be better -- certainly clearer -- to keep this entirely encapsulated in `anchoring/pdf`, and possibly expose a `reset` or `clearCache` function in the public interface of that module.